erpnext/setup/utils.py

```diff
@@ -62,19 +62,21 @@ def before_tests():
 	frappe.db.commit()
 
 @frappe.whitelist()
-def get_exchange_rate(from_currency, to_currency):
+def get_exchange_rate(from_currency, to_currency, posting_date=None):	
 	if from_currency == to_currency:
 		return 1
 	
-	exchange = "%s-%s" % (from_currency, to_currency)
-	value = flt(frappe.db.get_value("Currency Exchange", exchange, "exchange_rate"))
-
+	if not posting_date:
+		posting_date = frappe.utils.nowdate()
+		
+	value = flt(frappe.db.get_value("Currency Exchange", 
+		{"from_currency": from_currency, "to_currency": to_currency,
+			"date": posting_date}, "exchange_rate"))
```

Comment:
What about the case when a user wants to use a "fixed" exchange rate? The code should also look for a Currency Exchange record with a blank date. Or did I miss a discussion? @nabinhait what's your opinion?